### PR TITLE
feat: migrate over @fuels/assets package into the TS SDK

### LINKED ISSUE #1541: Export Asset/Networks Config Interface
Provide an interface and a JSON with Network/Assets metadata.

 - Could the JSON `live` in a CDN?
 - Should we have a `fetch` utility?

This is where it is today:
 - https://github.com/FuelLabs/fuels-npm-packs/tree/main/packages/assets

The idea is not to migrate it fully but only provide types and utilities around it.

### PR DESCRIPTION
**Pull request by Dhaiwat10**
This PR migrates the [`@fuels/assets`](https://github.com/fuellabs/fuels-npm-packs/tree/main/packages/assets) npm package to the TS SDK repo.

Todo:
- [x] Someone needs to add the AWS credentials to the TS SDK repo. These are required for the 'Upload assets to S3' workflow added with this PR. cc @luizstacio 

Closes #1541

### CODE REVIEW

**Comment by luizstacio** [comment]
@Dhaiwat10 the credentials are already setup on the CI.

**Comment by Dhaiwat10** [comment]
@luizstacio Oh didn't know, thanks! :)

**Review by arboleya — COMMENTED** [review]
Should we add some tests to this?

**Comment by Dhaiwat10** [comment]
Adding tests and also renaming some of the functions while I'm at it. Marking as a draft until ready

File: packages/account/src/providers/assets/utils/network.test.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * @group node
+ */
+describe('Network Utils', () => {
+  test('getDefaultChainId', async () => {
+    expect(getDefaultChainId('ethereum')).toBe(11155111);
+    expect(getDefaultChainId('fuel')).toBe(0);
+  })
+
+  test('getAssetWithNetwork - Ethereum', async () => {
```
Comment (nedsalk) [inline packages/account/src/providers/assets/utils/network.test.ts:15]:
```suggestion
test('getAssetNetwork - Ethereum', async () => {
```

File: packages/account/src/providers/assets/utils/network.test.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const asset = assets[0] as Asset
+    const assetEth = getAssetNetwork({ asset, networkType: 'ethereum', chainId: CHAIN_IDS.eth.sepolia })
+    expect(assetEth).toEqual({
+      type: 'ethereum',
+      chainId: CHAIN_IDS.eth.sepolia,
+      decimals: 18,
+    })
+  })
+
+  test('getAssetWithNetwork - Fuel', async () => {
```
Comment (nedsalk) [inline packages/account/src/providers/assets/utils/network.test.ts:25]:
```suggestion
  test('getAssetNetwork - Ethereum', async () => {
```

File: packages/account/src/providers/assets/utils/network.test.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    const assetFuel = getAssetNetwork({ asset, networkType: 'fuel', chainId: CHAIN_IDS.fuel.beta5 })
+    expect(assetFuel).toEqual({
+      type: 'fuel',
+      chainId: CHAIN_IDS.fuel.beta5,
+      decimals: 9,
+      assetId: '0x0000000000000000000000000000000000000000000000000000000000000000',
+    })
+  })
+
+  test('getAssetWithNetwork - invalid network', async () => {
```
Comment (nedsalk) [inline packages/account/src/providers/assets/utils/network.test.ts:36]:
```suggestion
  test('getAssetNetwork - invalid network', async () => {
```

File: packages/account/src/providers/assets/utils/network.test.ts (hunk 1 above)
Comment (nedsalk) [reply packages/account/src/providers/assets/utils/network.test.ts:15]: Wait, all my suggestions were based on you using `getAssetNetwork` but I just realized that `getAssetWithNetwork` also exists... Which one should  these tests be testing? :D

Comment (Dhaiwat10) [reply packages/account/src/providers/assets/utils/network.test.ts:15]: Ah my bad there was a mixup. Fixed now lol

**Review by arboleya — APPROVED** [review]

**Review by nedsalk — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**